client/app/pages/tournament-settings/setting-edit/setting-edit.component.ts

```diff
@@ -12,12 +13,37 @@ export class SettingEditComponent implements OnInit {
   @Input() setting: any
     public loading: boolean = true;
 
-  constructor(private tournamentSettingsService: TournamentSettingsService, private notification: Notifications) { }
+  constructor(private tournamentSettingsService: TournamentSettingsService, private notification: Notifications, private tournamentStatusService: TournamentStatusService) { }
 
   ngOnInit() {
   }
 
   save(){
+      if(this.setting.name === 'tournamentMatch'){
+        this.matchSave()
+      }else{
+        this.settingSave()
+    }
+  }
+
+  matchSave(){
+    if(isNaN(this.setting.value) || this.setting.value < 0 || this.setting.value > 5000){
```

Comment:
That's an input on the html side, So the component is getting the value from the outside

Reply:
So this check should be in the service itself and return an error if it is an illegitimate value?

Reply:
This is omitted after the merge? isnt it?